model/example_domains_test.go

```diff
@@ -0,0 +1,492 @@
+package model_test
+
+import (
+	"fmt"
+
+	"github.com/nextmv-io/sdk/model"
+)
+
+func ExampleNewDomains_first() {
+	domains := model.NewDomains()
+
+	fmt.Println(domains.Len())
+	fmt.Println(domains.Empty())
+
+	// Output:
+	// 0
+	// true
+}
+
+func ExampleNewDomains_second() {
+	domains := model.NewDomains(
+		model.Singleton(0),
+		model.Singleton(1),
+	)
+
+	fmt.Println(domains.Len())
+	fmt.Println(domains.Domain(0).Value())
+	fmt.Println(domains.Domain(1).Value())
+
+	// Output:
+	// 2
+	// 0 true
+	// 1 true
+}
+
+func ExampleRepeat_first() {
+	domains := model.Repeat(0, model.NewDomain())
+
+	fmt.Println(domains.Len())
+	fmt.Println(domains.Empty())
+
+	// Output:
+	// 0
+	// true
+}
+
+func ExampleRepeat_second() {
+	domains := model.Repeat(1, model.NewDomain())
+
+	fmt.Println(domains.Len())
+	fmt.Println(domains.Empty())
+
+	// Output:
+	// 1
+	// true
+}
+
+func ExampleRepeat_third() {
+	domains := model.Repeat(2, model.Singleton(1))
+
+	fmt.Println(domains.Len())
+	fmt.Println(domains.Domain(0).Value())
+	fmt.Println(domains.Domain(1).Value())
+
+	// Output:
+	// 2
+	// 1 true
+	// 1 true
+}
```

Comment:
Here is an example of how to condense the three examples into a single test. I think it is nice and compact and guides the reader to understand what domain they are seeing in the output. This suggestion can be applied to the other cases as well, where the domain "id" is printed alongside the result.

```suggestion
// Domains behave differently when repeating empty or integer domains.
func ExampleRepeat() {
	// An empty domain is repeated 0 times.
	d1 := model.Repeat(0, model.NewDomain())
	fmt.Println("d1: ", d1.Len())
	fmt.Println("d1: ", d1.Empty())
	
	// An empty domain is repeated 1 time.
	d2 := model.Repeat(1, model.NewDomain())
	fmt.Println("d2: ", d2.Len())
	fmt.Println("d2: ", d2.Empty())
	
	// A non-empty domain is repeated multiple times.
	d3 := model.Repeat(2, model.Singleton(1))
	fmt.Println("d3: ", d3.Len())
	fmt.Println("d3: ", d3.Domain(0).Value())
	fmt.Println("d3: ", d3.Domain(1).Value())
	// Output:
	// d1: 0
	// d1: true
	// d2: 1
	// d2: true
	// d3: 2
	// d3: 1 true
	// d3: 1 true
}
```